src/main/cliapp/src/service/TableStateService.js

```diff
@@ -0,0 +1,177 @@
+					"Updated By", "Date Updated", "Date Created", "Created By",
+			],
+			filters: {
+				obsoleteFilter: {
+					obsolete: { queryString: "false", tokenOperator: "OR" }
+				},
+					dataProviderFilter: {
+							"dataProvider.abbreviation": { queryString: "FB", tokenOperator: "AND" },
+							"dataProvider.fullName": { queryString: "FB", tokenOperator: "AND" },
+							"dataProvider.shortName": { queryString: "FB", tokenOperator: "AND" }
+					}
+			},
+			isFirst: false,
+			tableKeyName: "DiseaseAnnotations",
+			tableSettingsKeyName: "DiseaseAnnotationsTableSettings"
+		}
+	},
+	ZFINStaff:	{
+		DiseaseAnnotations: {
+			page: 0,
+			first: 0,
+			rows: 50,
+			multiSortMeta: [],
+			selectedColumnNames: [
+					"Subject", "Experiments", "Disease Relation", "Disease",
+					"Evidence Code", "Reference", "Inferred Gene", "Inferred Allele",
+					"Data Provider", "Updated By", "Date Updated", "Created By",
+					"Date Created", "Obsolete"
+			],
+			filters: {
+					dataProviderFilter: {
+							"dataProvider.abbreviation": { queryString: "ZFIN", tokenOperator: "AND" },
+							"dataProvider.fullName": { queryString: "ZFIN", tokenOperator: "AND" },
+							"dataProvider.shortName": { queryString: "ZFIN", tokenOperator: "AND" }
+					}
+			},
+			isFirst: false,
+			tableKeyName: "DiseaseAnnotations",
+			tableSettingsKeyName: "DiseaseAnnotationsTableSettings"
+		}
+	},
+	MGIStaff:	{
+		DiseaseAnnotations: {
+			page: 0,
+			first: 0,
+			rows: 50,
+			multiSortMeta: [],
+			selectedColumnNames: [
+					"Internal", "Obsolete", "Subject", "Disease Relation",
+					"Negated", "Disease", "Reference", "Evidence Code",
+					"Related Notes", "Inferred Gene", "Inferred Allele", "Data Provider",
+					"Updated By", "Date Updated", "Created By", "Date Created",
+			],
+			filters: {
+					dataProviderFilter: {
+							"dataProvider.abbreviation": { queryString: "MGI", tokenOperator: "AND" },
+							"dataProvider.fullName": { queryString: "MGI", tokenOperator: "AND" },
+							"dataProvider.shortName": { queryString: "MGI", tokenOperator: "AND" }
+					},
+					obsoleteFilter: {
+							obsolete: { queryString: "false", tokenOperator: "OR" }
+					}
+			},
+			isFirst: false,
+			tableKeyName: "DiseaseAnnotations",
+			tableSettingsKeyName: "DiseaseAnnotationsTableSettings"
+		}
+	}
+}
+
+export function getModTableState(table){
+	const oktaToken = JSON.parse(localStorage.getItem('okta-token-storage'));
+	const mod = oktaToken?.accessToken?.claims?.Groups?.filter(group => group.includes("Staff"));
+	return modTableSettings[mod][table]; 
+};
+
+export function getDefaultTableState(table, defaultColumnNames, defaultVisibleColumns){
+	return {
+			page: 0,
+			first: 0,
+			rows: 50,
+			multiSortMeta: [],
+			selectedColumnNames: defaultVisibleColumns ? defaultVisibleColumns : defaultColumnNames,
+			filters: {},
+			isFirst: false,
+			tableKeyName: table,
+			tableSettingsKeyName: `${table}TableSettings`
+	}
+} 
```

Comment:
@adamgibs just wondering if there might be a better way to do this given that this file is going to get huge. I know this was the initial idea. Maybe just leave it for now but just thinking.

Reply:
Yeah, it will be massive once we have all the tables included. That said, if we ever have an API set up for this, this map could go away.